### DISCUSSION TITLE
Encountered many problems

### DISCUSSION THREAD

**Discussion by HarveyTeng** [shivammathur/homebrew-php#209] [Q&A / Help] [answered]
httpd: Syntax error on line 183 of /usr/local/etc/httpd/httpd.conf: Cannot load /usr/local/Cellar/php@7.0/7.0.33.reinstall/lib/httpd/modules/libphp7.so into server: dlopen(/usr/local/Cellar/php@7.0/7.0.33.reinstall/lib/httpd/modules/libphp7.so, 10): Library not loaded: /usr/local/opt/icu4c/lib/libicui18n.68.dylib\n  Referenced from: /usr/local/Cellar/php@7.0/7.0.33.reinstall/lib/httpd/modules/libphp7.so\n  Reason: image not found

**Comment by HarveyTeng** [comment]
And my device big sur   use brew installed apache 2.4 . When     sudo brew apache -k restart / stop

**Reply by shivammathur** [reply]
Reinstall PHP 7.0 and its dependencies

```
brew reinstall $(brew deps shivammathur/php/php@7.0) shivammathur/php/php@7.0
```

Then run

```
find $(brew --cellar)/php@7.0 -name 'libphp7.so'
```

It should return a path like `/usr/local/Cellar/php@7.0/7.0.33/lib/httpd/modules/libphp7.so`

Use this path in `httpd.conf` for loading `libphp7.so`

**Reply by HarveyTeng** [reply]
ha，also should problem ~

but，When run 
  brew reinstall $(brew deps shivammathur/php/php@7.0) shivammathur/php/php@7.0

end，
Warning: The post-install step did not complete successfully
You can try again using:
  brew postinstall shivammathur/php/php@7.0

**Reply by shivammathur** [reply]
Make sure your brew permissions are correct
```
sudo chown -R "$(id -un)":"$(id -gn)" $(brew --prefix)
```

then follow https://github.com/shivammathur/homebrew-php/discussions/209#discussioncomment-462777

**Reply by HarveyTeng** [reply]
I think my permissions are correct .because when i try brew install php@7.2 and apache run normally

**Reply by shivammathur** [reply]
Permissions might be wrong in PHP@7.0 directory, please run 
```
sudo chown -R "$(id -un)":"$(id -gn)" $(brew --prefix)
```

and provide complete logs of the following
```
brew reinstall $(brew deps shivammathur/php/php@7.0) shivammathur/php/php@7.0
brew link --force php@7.0
```
Restart terminal and run
```
php -v
```

**Reply by shivammathur** [reply]
Also make sure you have the latest version of XCode CLT installed
Run `brew config`, it will show the XCode and CLT version

To update the CLT version.
```
sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install
```

Then reinstall php and its dependencies

**Reply by HarveyTeng** [reply]
Hi Shivam Mathur .
My first step run  `sudo chown -R "$(id -un)":"$(id -gn)" $(brew --prefix)` 
And then `sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install`

Finaly,  `brew reinstall $(brew deps shivammathur/php/php@7.0) shivammathur/php/php@7.0
brew link --force php@7.0`

terminal log show ：
`Linking /usr/local/Cellar/php@7.0/7.0.33.reinstall... 25 symlinks created.

If you need to have this software first in your PATH instead consider running:
  echo 'export PATH="/usr/local/opt/php@7.0/bin:$PATH"' >> ~/.zshrc
  echo 'export PATH="/usr/local/opt/php@7.0/sbin:$PATH"' >> ~/.zshrc`

It seems to be correct until now。
Then open new terminal and run `php -v`  It's show `PHP 7.2.33 (cli) (built: Aug  7 2020 18:29:34) ( NTS )`
Why php7.2？ My mac os php version is 7.3.

and when i run  `apachectl -k restart `  log also show 

`sudo apachectl -k start
httpd: Syntax error on line 184 of /usr/local/etc/httpd/httpd.conf: Cannot load /usr/local/opt/php@7.0/lib/httpd/modules/libphp7.so into server: dlopen(/usr/local/opt/php@7.0/lib/httpd/modules/libphp7.so, 10): Library not loaded: /usr/local/opt/icu4c/lib/libicui18n.68.dylib\n  Referenced from: /usr/local/opt/php@7.0/lib/httpd/modules/libphp7.so\n  Reason: image not found`

And  run `brew services list` ：
`httpd   error   /Users/tengwei/Library/LaunchAgents/homebrew.mxcl.httpd.plist
php@7.0 error   /Users/tengwei/Library/LaunchAgents/homebrew.mxcl.php@7.0.plist
php@7.1 error    /Users/tengwei/Library/LaunchAgents/homebrew.mxcl.php@7.1.plist
php@7.2 stopped   `

**Comment by shivammathur** [comment, marked as answer]
Run this to switch to php@7.0
```
brew link --force --overwrite php@7.0
```

It seems icu4c is not updated on your system
Run `brew update` and then reinstall icu4c
```
brew reinstall icu4c
```
then check php

**Reply by HarveyTeng** [reply]
Thanks very much .
After several network interruptions, the reinstallation was finally successful, and the startup and operation have been normal.
Before that, I also tried to reinstall icu4c, but I also reported an error. I don’t know if it is caused by the installation sequence.
